extensions/amp-insticator/0.1/amp-insticator.js

```diff
@@ -0,0 +1,179 @@
+  }
+
+  /** @override */
+  layoutCallback() {
+    // get data attribute from the amp-insticator tag
+    const embedId = this.element.getAttribute('data-embed-id');
+    // store DOM elements
+    const insticatorContainer = this.element.querySelector('#insticator-container');
+    const embedIframe = this.iFrameElement = this.element.querySelector('#insticator-iframe');
+    const embedApp = this.createElement(this.element.ownerDocument, 'div', { id: 'app' });
+    const embedScript = this.createElement(this.element.ownerDocument, 'script', { type: 'text/javascript', src: `${this.url.embed}/embed-code/${embedId}.js` });
+
+    // append iframe
+    this.appendElement(this.element.querySelector('#insticator-embed'), embedIframe);
+
+    // append content to iframe
+    this.appendElement(embedIframe.contentWindow.document.body, embedApp);
+    this.appendElement(embedIframe.contentWindow.document.head, embedScript);
+
+    // append ads
+    this.getRequest(`${this.url.ads}/test/ad_settings/${embedId}.js`, (ads) => this.appendAds(ads));
+
+    // apply custom styles
+    setStyles(insticatorContainer, { 'text-align': 'center' });
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER;
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iFrameElement) {
+      removeElement(this.iFrameElement);
+      this.iFrameElement= null;
+    }
+    return true; // Call layoutCallback again.
+  }
+
+
+  // ----- CUSTOM FUNCTIONs ----- //
+  /**
+   * Make get request to get ads data.
+   * @param {!string} file
+   * @param {function(Object)} callback
+   */
+  getRequest(file, callback) {
+    const xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = () => xhr.readyState == 4 && xhr.status == 200 ? callback(JSON.parse(xhr.responseText)) : null;
+    xhr.open('GET', file, true);
+    xhr.send();
+  }
+
+  /**
+   * Create markup structure.
+   * @param {!string} domString
+   * @return {!Element}
+   */
+  createTemplate(domString) {
+    const template = document.createElement('template');
+    template.innerHTML = domString;
+    return template.content;
+  }
+
+  /**
+   * Create an element with attributes.
+   * @param {!Element} location
+   * @param {!string} el
+   * @param {!Object} attrs
+   * @return {!Element}
+   */
+  createElement(location, el, attrs) {
+    const newEl = location.createElement(el);
+    Object.entries(attrs).forEach(attr => newEl.setAttribute(attr[0], attr[1]));
+    return newEl;
+  }
+
+  /**
+   * Append a node to a specified parent node.
+   * @param {!Element} location
+   * @param {!Element} el
+   */
+  appendElement(location, el) {
+    location.appendChild(el);
+  }
+
+  /**
+   * Append ads.
+   * @param {!Object} ads
+   */
+  appendAds(ads) {
+    Object.entries(ads).forEach(ad => this.appendElement(this.element.querySelector(`#div-insticator-ad-${ad[0][ad[0].length -1]}`), this.createElement(this.element.ownerDocument, 'amp-ad', ad[1])));
```

Comment:
inserting amp-ad like this can cause page reflow. 
you will need to use attempChangeHeight to increase your element height and then append ads.

Reply:
We will address this and let you know when updated.